### ISSUE TITLE
Customized Processing module doesn't work

### ISSUE THREAD

**Issue by cssxn** [cuckoosandbox/cuckoo#2383]
I just wrote a demo for test. here is my code

```
from cuckoo.common.abstracts import Processing
from cuckoo.common.objects import File

class Test(Processing):

    def run(self):
        self.key = "test"

        data = {"test":"wdy"}
        return data
```

When a sample analysed I examined the data of report from `mongoDB` which didn't show the `test` key in that result data container.

**Comment by doomedraven** [comment]
`def run` is dead as far as i remember, try `on_complete`

**Comment by cssxn** [comment]
@doomedraven  I don't think so. Because others processing module in the `cuckoo/processing/* `  have `def run` function, you can check it

**Comment by doomedraven** [comment]
ah wait i was thinking about signatures, did you add your plugin to processing.conf? if no it won't run

**Comment by cssxn** [comment]
I see. thank u bro.

**Comment by cssxn** [comment]
@doomedraven  @RicoVZ 
still did not run.. the cuckoo log still says

```
 [cuckoo.core.plugins] DEBUG: Processing module filehealth not found in configuration file
```

I've added plugin to processing.conf

```
[filehealth]                                                         
enabled = yes  
```

cuckoo/processing/filehealth.py

```
class FileHealth(Processing):

    enabled = True

    def run(self):

        self.key = "filehealth"
```

**Comment by RicoVZ** [comment]
Hi @cssxn,

Thanks for posting an issue.

Implementing `run` is correct. The run method should return the results.
That message means it was added to a config, but it is an unknown config. Cuckoo uses an internal file to determine how a specific value in the config should be interpreted.

For your processing module, you should add it to. https://github.com/cuckoosandbox/cuckoo/blob/master/cuckoo/common/config.py#L512

A minimum is the `enabled` key.

```python
"filehealth": {
    "enabled": Boolean(True),
}
```

**Comment by cssxn** [comment]
@RicoVZ  Thanks, it works!

**Comment by bitninjaio** [comment]
It should be highlighted in the doc (https://cuckoo.readthedocs.io/en/latest/customization/processing/). I was looking for this like 2 hours :-) Based on the doc it is enough to add this section to the user level config. Thanks for the solution anyway.
